### ISSUE TITLE
GCS crashes on startup in linux (next branch)

### ISSUE THREAD

**Issue by thunderdan** [d-ronin/dRonin#829]
I've seen this for a couple of days on my machine.  The gcs won't start and the error message is below.

ASSERT failure in QVector<T>::at: "index out of range", file /home/dmorgan/code/dronin-readonly/tools/Qt5.5.1/5.5/gcc_64/include/QtCore/qvector.h, line 393

I wouldn't think anything is wrong with the qvector.h file providied w/ Qt so probably something in the usage of it is wrong.

A dump file is written to disk so i can upload if that helps.  I could also back track to see when the issue was introduced if you like.

**Comment by thunderdan** [comment]
I tried building two earlier commits and both crash on startup for me in the same manor.

Fail - commit 54765589fa8d7050e8244334f73e9d02539aadfa
Fail - commit d29298611ed4a9599b61481d5f2c0bfcb5689966

Here is the output I get when lauching 

<pre>
./build/ground/gcs/bin/drgcs
"fusion"
""
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
Starting the Udev client
Starting the Udev client
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
"addOverrideAction: action already registered for context when registering ''"
UavTalkRelay SERVER listening on  QHostAddress("0.0.0.0") 2000
qt.network.ssl: QSslSocket: cannot resolve SSLv2_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv2_server_method
QObject::connect: Cannot connect (null)::stateChanged(QNetworkSession::State) to QNetworkReplyHttpImpl::_q_networkSessionStateChanged(QNetworkSession::State)
QObject::connect: Cannot connect (null)::stateChanged(QNetworkSession::State) to QNetworkReplyHttpImpl::_q_networkSessionStateChanged(QNetworkSession::State)
QObject::connect: Cannot connect (null)::stateChanged(QNetworkSession::State) to QNetworkReplyHttpImpl::_q_networkSessionStateChanged(QNetworkSession::State)
QObject::connect: Cannot connect (null)::stateChanged(QNetworkSession::State) to QNetworkReplyHttpImpl::_q_networkSessionStateChanged(QNetworkSession::State)
registerShortcut: action already registered (id "GCS.Mode.Welcome" .
Loaded stylesheet: "/home/dmorgan/code/dRonin-readonly/build/ground/gcs/share/stylesheets" "default"
ASSERT failure in QVector<T>::at: "index out of range", file /home/dmorgan/code/dronin-readonly/tools/Qt5.5.1/5.5/gcc_64/include/QtCore/qvector.h, line 393
Dump file was written to: /tmp/dr_next_20160213_429c4b76/6750eee2-7939-3619-765113a8-43b24421.dmp
</pre>

**Comment by mlyle** [comment]
Try resetting the ground configuration -- maybe you've gotten that into an invalid state -- `./build/ground/gcs/bin/drgcs -r`

If this doesn't work, can you try running from this binary distribution?  http://jar.lyle.org:8080/job/dronin/263/Nodes=master/artifact/build/artifacts/dronin-20160322-2f65e4b7.tar.xz

**Comment by tracernz** [comment]
Running with the debugger from Qt creator will show you exactly where this is coming from.

**Comment by thunderdan** [comment]
Before I tried your other suggestions, I did `make gcs_clean` followed by `make gcs`.  The problem goes away and  I can run the GUI again now.  

As a side note: Running dronin from the binary distro from January has always been fine.  The about dialog displays "Release-20160120.3"

Do the settings for gcs built from src get stored in a separate location than the settings when installed from binary?

I'll close the issue and if I see any issues like this again I'll try to get more info with the debugger.
